google/resource_sql_database_instance_test.go

```diff
@@ -629,6 +629,38 @@ func testAccCheckGoogleSqlDatabaseRootUserDoesNotExist(t *testing.T, instance st
 	}
 }
 
+func TestAccSqlDatabaseInstance_PointInTimeRecoveryEnabled(t *testing.T) {
+	t.Parallel()
+
+	masterID := randInt(t)
+
+	vcrTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccSqlDatabaseInstanceDestroyProducer(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testGoogleSqlDatabaseInstance_PointInTimeRecoveryEnabled(masterID, true),
+				Check:  resource.TestCheckResourceAttr("google_sql_database_instance.instance", "point_in_time_recovery", "true"),
```

Comment:
This check isn't necessary- if the value in state didn't match the value in config, this test step would fail with a non-empty plan. The import step below also checks that we properly read the value from the API, so we aren't just blindly putting the value from the config into state.

Reply:
Appreciate your patience and inputs 🙏 . I would like to clarify to prevent such mistakes in the future. 
Isn't Check function executed after Config is applied? In that case, the Check ensures that value in the state is same as we set in the config. In the second test step I'm changing the value from true -> false, and again ensuring that it was changed properly in the state. 
If this is covered by the import step - I didn't know that.

Reply:
Eh it's not a mistake, just a bit redundant is all. The check does ensure that the value in the state is the same as the config. However, the overall test framework runs a `terraform plan` after every TestStep to ensure that the plan is empty. So if the value in the config didn't match the state, that would show a diff in the plan, and the TestStep would fail automatically.

Reply:
Got it. Thank you, Dana, for the explanation and steel nerves 🙂 Will remove it.